MTM-64669 Measurement series aggregation functions (#3846)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-2025-356-0-measurement-series-aggregation-functions.md

---
date: ''

> title: New parameter function added to measurement series endpoint

Reviewer: ```suggestion
title: New parameter added to measurement series endpoint
```

change_type:
  - value: change-QHu1GdukP
    label: Feature
product_area: Platform services
component:
  - value: component-JlFdtOPva
    label: REST API
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-64669
version: 2025.356.0
---
A new query parameter `function` has been added to the measurement series endpoint. This parameter allows to specify
aggregation functions to be calculated on the measurement series data per specified aggregation interval.

> The supported aggregation functions are: `min`, `max`, `avg`, `sum`, `count`, `stdDevPop`, and `stdDevSamp`.

Reviewer 2: There should be an information that a tenant must have time series measurements enabled to get this feature

Author: Added.

> The supported aggregation functions are: `min`, `max`, `avg`, `sum`, `count`, `stdDevPop` and `stdDevSamp`.

Reviewer: ```suggestion
The supported aggregation functions are: `min`, `max`, `avg`, `sum`, `count`, `stdDevPop`, and `stdDevSamp`.
```

> If the parameter is not provided functions `min` and `max` will be calculated by default.

Reviewer: ```suggestion
If the parameter is not provided, the functions `min` and `max` will be calculated by default.
```

{{< c8y-admon-info >}}

> If you want to calculate multiple functions at once, you must specify the parameter multiple times, e.g. `function=min&function=max&function=avg`.

Reviewer: ```suggestion
If you want to calculate multiple functions at once, you must specify the parameter multiple times, for example,  `function=min&function=max&function=avg`.
```

> If you want to calculate multiple functions at once, you must specify the parameter multiple times, for example,  `function=min&function=max&function=avg`.

Reviewer: ```suggestion
If you want to calculate multiple functions at once, you must specify the parameter multiple times, for example, `function=min&function=max&function=avg`.
```

---
date: ''

> title: New parameter added to measurement series endpoint

Reviewer 3: Perhaps update to 'Aggregation Improvements'

Author: Renamed

    label: Feature
product_area: Platform services
component:
  - value: component-JlFdtOPva
    label: REST API
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-64669
version: 2025.356.0
---
A new query parameter `function` has been added to the measurement series endpoint. This parameter allows to specify
aggregation functions to be calculated on the measurement series data per specified aggregation interval.
The supported aggregation functions are: `min`, `max`, `avg`, `sum`, `count`, `stdDevPop`, and `stdDevSamp`.
If the parameter is not provided, the functions `min` and `max` will be calculated by default.

{{< c8y-admon-info >}}
If you want to calculate multiple functions at once, you must specify the parameter multiple times, for example, `function=min&function=max&function=avg`.
{{< /c8y-admon-info >}}

{{< c8y-admon-caution >}}
This parameter can only be specified when an aggregation interval is provided using `aggregationType` or `interval` parameter.
Providing the `function` parameter without an aggregation interval will result in a bad request error.
This parameter only takes effect for the timeseries persistence of measurements.

> For legacy persistence always `min` and `max` functions are calculated.

Reviewer: ```suggestion
For legacy persistence, the `min` and `max` functions are always calculated.
```
